app/src/main/java/com/medmapper/v33001/repository/MedicineRepository.kt

```diff
@@ -18,7 +19,13 @@ class MedicineRepository(private val medicineDAO: MedicineDAO) {
     // long running database work off the main thread
     @Suppress("RedundantSuspendModifier")
     @WorkerThread
-    suspend fun insert(medicine: Medicine) {
-        medicineDAO.insert(medicine)
+    suspend fun insertMedicine(medicine: Medicine) {
```

Comment:
Added Try Catch here - suggesting that there shouldn't be any errant entries in the database.

Reply:
Logging is good.